storybook/src/stories/UseDeepCompareEffect/API/API.tsx

```diff
@@ -0,0 +1,35 @@
+import throttle from 'lodash.throttle'
+import React, { useRef, useState } from 'react'
+import { useDeepCompareEffect } from 'too-many-hooks'
+import { COUNTRY_CODES } from './constants'
+
+interface Props {
+  firstName: string
+  countryKey: keyof typeof COUNTRY_CODES
+}
+
+const API: React.FC<Props> = ({ firstName, countryKey }) => {
+  const [predictedAge, setPredictedAge] = useState<string>()
+
+  const getThrottledPrediction = useRef(
+    throttle(
+      (params: { name: string; country_id: typeof COUNTRY_CODES[keyof typeof COUNTRY_CODES] }) =>
+        fetch('https://api.agify.io?' + new URLSearchParams(params))
```

Comment:
This fetch is happening twice with every change. Seems that the throttle is calling the most recent unblocked callback after it's wait duration.

Reply:
Not sure what you mean here. It is behaving as I expect, querying at most every 2 seconds. If I type "Dawson" it queries for "D" and then 2 seconds later "Dawson". Are you seeing something different?

Reply:
If I type "D" (or any string) it'll query once and then 2 seconds later perform the same query, unless I interrupt the wait with another string. But in any case if I wait the 2 seconds then it'll perform the most recent query again.